### PR TITLE
Add dotnet package search command

### PR DESCRIPTION
**Pull request by Nigusu-Allehu**
This PR will
Fixes issues: https://github.com/NuGet/Home/issues/6060  
- add a new command `dotnet package search`
- The spec of the new command is found here : https://github.com/NuGet/Home/pull/12886

### CODE REVIEW

File: src/Cli/dotnet/commands/dotnet-package/search/NugetSearchApiRequest.cs [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                    take: _take ?? 20,
+                    NullLogger.Instance,
+                    CancellationToken.None
+                    )), source));
+                Console.WriteLine($"{source.Name}4");
+            }
+            Console.WriteLine("wow " + taskList.Count().ToString() );
+            foreach (var taskItem in taskList)
+            {
+                Console.WriteLine("WTF");
```
Comment (skofman1) [inline src/Cli/dotnet/commands/dotnet-package/search/NugetSearchApiRequest.cs:67]: You should squash it before merging, so it's not part of the commit history for the repo :)

File: src/Cli/dotnet/commands/dotnet-package/search/LocalizableStrings.resx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <data name="CommandDescription" xml:space="preserve">
+    <value>Search for a package in nuget.org. </value>
```
Comment (danmoseley) [inline src/Cli/dotnet/commands/dotnet-package/search/LocalizableStrings.resx:121]:
```suggestion
    <value>Search for a package in nuget.org.</value>
```

File: src/Cli/dotnet/commands/dotnet-package/search/LocalizableStrings.resx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+    <value>A list of package sources to search.</value>
+  </data>
+  <data name="TakeArgumentName" xml:space="preserve">
+    <value>Take</value>
+  </data>
+  <data name="TakeDescription" xml:space="preserve">
+    <value>The number of results to return. The default value is 20.</value>
+  </data>
+  <data name="Warning_HttpServerUsage" xml:space="preserve">
+    <value>You are running the '{0}' operation with an 'HTTP' source, '{1}' Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source</value>
```
Comment (danmoseley) [inline src/Cli/dotnet/commands/dotnet-package/search/LocalizableStrings.resx:157]:
```suggestion
    <value>You are running the '{0}' operation with an 'HTTP' source, '{1}'. Non-HTTPS access will be removed in a future version. Consider migrating to an 'HTTPS' source.</value>
```

File: src/Cli/dotnet/commands/dotnet-package/search/NugetSearchApiRequest.cs [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+                throw new Exception("Invalid source " + source);
+            }
+        }
+
+        private void WarnForHTTPSources(IList<PackageSource> packageSources)
+        {
+            List<PackageSource> httpPackageSources = null;
+            foreach (PackageSource packageSource in packageSources)
+            {
+                if (packageSource.IsHttp && !packageSource.IsHttps)
```
[the remark this answers is not in the document]
Comment (Nigusu-Allehu) [reply src/Cli/dotnet/commands/dotnet-package/search/NugetSearchApiRequest.cs:135]: Noted, thank you!

**Comment by Nigusu-Allehu** [comment]
I have decided to move the implementation to the nuget.client repo side
